libraries/opensk/src/ctap/mod.rs

```diff
@@ -16,9 +16,8 @@ pub mod apdu;
 mod client_pin;
 pub mod command;
 mod config_command;
-mod credential_id;
 mod credential_management;
-mod crypto_wrapper;
+pub mod crypto_wrapper;
```

Comment:
`PrivateKey` should indeed be moved out to `api`. That's the only type that needs to move.

Reply:
What about the others we expose and use?

- `Secret`
- `CredentialProtectionPolicy`
- `aes256_cbc_decrypt`, `aes256_cbc_encrypt`
- `cbor_read`, `cbor_write`

And more for vendor commands from the other PR:

- `Ctap2StatusCode`
- `check_user_presence`
- several other types from `data_formats::`

I think this is the downside of more freedom in `Env`: You need to understand some CTAP concepts to implement credential wrapping, vendor commands etc.

Reply:
Actually the best way to figure out what should be in the API, is just to compile the documentation with `cargo doc` and go through all the API looking at the exposed types and see if they make sense.

Reply:
Question is if all `data_formats` should split and some of it moves to API?

The other types and functions like `Ctap2StatusCode` and `check_user_presence` are exposed so that we can implement custom vendor functions. We could reimplement all of them, and have trivial conversions into the internal types, but that seems like a bad idea.

If we go the `cargo doc` route and plan our API, I think that's a bigger change than what is proposed here. So just moving `PrivateKey` is one option for now?

Reply:
> So just moving PrivateKey is one option for now?

Yes, that's exactly what my initial comment was about. It's fine to not fix everything, but PRs should not add new things to fix.

The rest is just the guidelines to decide what should be in the API or not. Such that future PRs follow that guideline.

Reply:
Right, done!